### PR TITLE
enum_domain: Cleanup and support non-Meterpreter sessions

### PR DESCRIPTION
**Pull request by bcoles**
Note: This requires #16952 to be merged first.

---

Resolves Rubocop violations.

Adds `Notes` module meta information.

Adds support for non-Meterpreter sessions.

### CODE REVIEW

File: documentation/modules/post/windows/gather/enum_domain.md [hunk 1]
```diff
@@ -10,34 +10,36 @@ This module identifies the primary domain via the registry. The registry value u
...
 1. Do: `run`
-1. You should information on the computer's domain
+1. You should receive Active Directory domain information
 
 ## Options
 
 ## Scenarios
 
-### Windows 2012 DC
+### Windows 2016 with Windows 2008 SP1 DC
```
[the remark this answers is not in the document]
Comment (bcoles) [reply documentation/modules/post/windows/gather/enum_domain.md:19]: I don't see why not.

**Review by gwillcox-r7 — CHANGES_REQUESTED** [review]

File: modules/post/windows/gather/enum_domain.rb [hunk 2]
```diff
@@ -30,40 +35,52 @@ def initialize(info = {})
...
-          host: dc_ip,
-          name: dom_info[0],
-          info: "Domain controller for #{dom_info[1]}"
-        })
-      else
-        print_good("FOUND Domain Controller: #{dom_info[0]}")
-      end
+    domain = get_domain_name
+
+    fail_with(Failure::Unknown, 'Could not retrieve domain name. Is the host part of a domain?') unless domain
```
[the remark this answers is not in the document]
Comment (gwillcox-r7) [reply modules/post/windows/gather/enum_domain.rb:55]: Will be fixed in an upcoming commit.

File: modules/post/windows/gather/enum_domain.rb [hunk 3]
```diff
@@ -30,40 +35,52 @@ def initialize(info = {})
...
+      update: :unique_data
+    )
+
+    netbios_domain_name = domain.split('.').first.upcase
+
+    print_good("Domain NetBIOS Name: #{netbios_domain_name}")
+
+    domain_controller = get_primary_domain_controller
+
+    fail_with(Failure::Unknown, 'Could not retrieve domain controller name') unless domain_controller
```
[the remark this answers is not in the document]
Comment (gwillcox-r7) [reply modules/post/windows/gather/enum_domain.rb:72]: Will be fixed in an upcoming commit.

**Comment by gwillcox-r7** [comment]
With the changes listed above this should be good to go and works well in my tests. Will land now.

**Comment by gwillcox-r7** [comment]
# Release Notes
The `enum_domain` script has been updated to support Powershell and Shell sessions and its documentation and code have been cleaned up.

File: modules/post/windows/gather/enum_domain.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (gwillcox-r7) [reply modules/post/windows/gather/enum_domain.rb]: Fixed in https://github.com/rapid7/metasploit-framework/commit/7869e9229858e9e8de11aeea2685c50c812c01bd

File: modules/post/windows/gather/enum_domain.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (gwillcox-r7) [reply modules/post/windows/gather/enum_domain.rb]: Fixed in https://github.com/rapid7/metasploit-framework/commit/7869e9229858e9e8de11aeea2685c50c812c01bd

**Comment by gwillcox-r7** [comment]
PR was merged in https://github.com/rapid7/metasploit-framework/commit/bf19b722b135035d5ac135306c8bc504654af915, sorry for the mess 😞

**Pull request closed without merging**